internal/lookout/ui/src/services/JobService.ts

```diff
@@ -1,56 +1,46 @@
 import { LookoutApi, LookoutJobInfo } from '../openapi'
+import { reverseMap } from "../utils";
 
 export type JobInfoViewModel = {
   jobId: string
   queue: string
   owner: string
   jobSet: string
   submissionTime: string
-  jobState: string
+  jobState: JobStateViewModel
 }
 
-function jobInfoToViewModel(jobInfo: LookoutJobInfo): JobInfoViewModel {
-  const jobId = jobInfo.job?.id ?? "-"
-  const queue = jobInfo.job?.queue ?? "-"
-  const owner = jobInfo.job?.owner ?? "-"
-  const jobSet = jobInfo.job?.jobSetId ?? "-"
-  const submissionTime = (jobInfo.job?.created ?? new Date()).toLocaleString()
-  const jobState = parseJobState(jobInfo)
-
-  return {
-    jobId: jobId,
-    queue: queue,
-    owner: owner,
-    jobSet: jobSet,
-    submissionTime: submissionTime,
-    jobState: jobState,
-  }
+export enum JobStateViewModel {
```

Comment:
Could you potentially combine the 2 enums? Maybe using union of string literals as a type would make it work better (I haven't tried).